### PR TITLE
Add listener detach functionality

### PR DESCRIPTION
**Pull request by dilanSachi**
## Purpose
$subject

## Checklist
- [x] Linked to an issue
- [x] Updated the changelog
- [x] Added tests

Fixes https://github.com/ballerina-platform/ballerina-standard-library/issues/2211

### CODE REVIEW

File: native/src/main/java/io/ballerina/stdlib/kafka/api/KafkaServerConnector.java [hunk 1]
```diff
@@ -48,4 +48,12 @@
      */
     boolean immediateStop() throws KafkaConnectorException;
 
+    /**
+     * Stops the scheduled poll task.
+     *
+     * @return true if stopped successfully, false otherwise
+     * @throws KafkaConnectorException if error occurred while stopping the Kafka server connector
+     */
+    boolean stopPollingTask() throws KafkaConnectorException;
```
[the remark this answers is not in the document]
Comment (dilanSachi) [reply native/src/main/java/io/ballerina/stdlib/kafka/api/KafkaServerConnector.java]: Changed in https://github.com/ballerina-platform/module-ballerinax-kafka/pull/310/commits/b555729a5872ce789987d78c0c11ca8cbb0f516e

File: native/src/main/java/io/ballerina/stdlib/kafka/service/Unregister.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
+package io.ballerina.stdlib.kafka.service;
```
Comment (shafreenAnfar) [inline native/src/main/java/io/ballerina/stdlib/kafka/service/Unregister.java:1]: License header.

Comment (dilanSachi) [reply native/src/main/java/io/ballerina/stdlib/kafka/service/Unregister.java:19]: Added https://github.com/ballerina-platform/module-ballerinax-kafka/pull/310/commits/3450710e0c2005fa6f003e297f375b5c641d540f

File: native/src/main/java/io/ballerina/stdlib/kafka/impl/KafkaRecordConsumer.java [hunk 3]
```diff
@@ -177,4 +177,21 @@ public void immediateStopConsume() {
...
+    /**
+     * Stops Kafka consumer polling cycles and forcefully shutdowns scheduled thread pool.
+     */
+    public void stopScheduledPollTask() {
+        closed.set(true);
+        this.kafkaConsumer.wakeup();
+        this.pollTaskFuture.cancel(true);
+        this.executorService.shutdownNow();
+        try {
+            this.executorService.awaitTermination(stopTimeout, TimeUnit.MILLISECONDS);
```
Comment (shafreenAnfar) [inline native/src/main/java/io/ballerina/stdlib/kafka/impl/KafkaRecordConsumer.java:190]: Lets make 30 seconds.

Comment (dilanSachi) [reply native/src/main/java/io/ballerina/stdlib/kafka/impl/KafkaRecordConsumer.java:190]: Changed in https://github.com/ballerina-platform/module-ballerinax-kafka/pull/310/commits/36c9dd547c14677dee1c4c474f7bc28846d084f1

**Review by shafreenAnfar — APPROVED** [review]
LGTM!

**Pull request merged**